android/src/com/frostwire/android/gui/fragments/BrowsePeerFragment.java

```diff
@@ -485,6 +688,114 @@ private void changeSelectedRadioButton(byte currentFileType, byte nextFileType)
             nextButton.setChecked(true);
             nextButton.callOnClick();
         }
+
+    }
+
+    public boolean inSpecialState() {
+        return fragmentState != ViewState.NORMAL;
+    }
+
+    public void endSpecialState() {
+        if (fragmentState == ViewState.CHECK) {
+            endCheckState();
+        } else if (fragmentState == ViewState.FILTERING) {
+            endFilteringState();
+        } else if (fragmentState == ViewState.FILTERED) {
+            endFilteredState();
+        }
+    }
+
+    public void endCheckState() {
+        fragmentState = beforeCheckState;
+        if (beforeCheckState == ViewState.FILTERED) {
+            ((MainActivity) getActivity()).swapDrawerForBack();
+        }
+        swipe.setOnSwipeListener(swipeListener);
+        adapter.setCheckState(false);
+        updateViewsToState();
+    }
+
+    public void enterCheckState() {
```

Comment:
Yes, this evolved (mutated rather :p) from just one Checked state, through Checked and Filtering to this. (I didn't get for some reason (now it seems clear) what the designs were representing and had to change how it works)